includes/plugins/woocommerce/my-account/class-my-account-ui-v1.php

```diff
@@ -119,5 +135,305 @@ public static function my_account_menu_items( $items ) {
 		unset( $items['customer-logout'] );
 		return $items;
 	}
+
+	/**
+	 * Remove required fields from the account settings form.
+	 *
+	 * @param array $required_fields The required fields.
+	 * @return array The required fields.
+	 */
+	public static function account_settings_required_fields( $required_fields ) {
+		unset( $required_fields['account_display_name'] );
+		return $required_fields;
+	}
+
+	/**
+	 * Display a series of modals to request account deletion.
+	 */
+	public static function delete_account_modal() {
+		$user = \wp_get_current_user();
+
+		// Only if the user is logged in and a reader.
+		if ( ! \is_user_logged_in() || ! Reader_Activation::is_user_reader( $user ) ) {
+			return;
+		}
+
+		// If the user has clicked the button from the delete account email, show the confirmation modal.
+		$delete_account_form = filter_input( INPUT_GET, WooCommerce_My_Account::DELETE_ACCOUNT_FORM, FILTER_SANITIZE_FULL_SPECIAL_CHARS );
+		if ( ! empty( $delete_account_form ) ) {
+			return self::delete_account_confirmation_modal();
+		}
+
+		$active_subscriptions     = WooCommerce_Connection::get_active_subscriptions_for_user( $user->ID );
+		$newsletter_subscriptions = method_exists( 'Newspack_Newsletters_Subscription', 'get_contact_lists' ) ? \Newspack_Newsletters_Subscription::get_contact_lists( $user->data->user_email ) : [];
```

Comment:
Noting that this will increase load time for the My Account dashboard page, as it performs a remote request to the ESP API.

It's suboptimal, but I wouldn't say it's something to be optimized in this PR.

Reply:
Here's how that might look: 2e51a2be

Reply:
Unless we want to augment the `get_data` method so that it can optionally return decoded data? No getting around the need to decode somewhere, though.